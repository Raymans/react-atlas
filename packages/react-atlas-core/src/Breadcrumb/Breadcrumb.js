import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class Breadcrumb extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const { className, style, children } = this.props;

    return (
      <ol
        styleName="breadcrumb"
        className={cx(className)}
        style={style}
        role="navigation"
        aria-label="breadcrumbs"
      >
        {children}
      </ol>
    );
  }
}

Breadcrumb.propTypes = {
  /** An object, array, or string of CSS classes to apply to Breadcrumb.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,
  /**
   * BreadcrumbItem instances.
   */
  "children": PropTypes.node
};

Breadcrumb.defaultProps = {
  "className": ""
};

export default Breadcrumb;
