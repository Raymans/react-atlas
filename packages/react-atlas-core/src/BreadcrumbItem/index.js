import { default as BreadcrumbItemCore } from "./BreadcrumbItem";
export { BreadcrumbItemCore };
