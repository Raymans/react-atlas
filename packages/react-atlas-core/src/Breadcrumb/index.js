import { default as BreadcrumbCore } from "./Breadcrumb";
export { BreadcrumbCore };
