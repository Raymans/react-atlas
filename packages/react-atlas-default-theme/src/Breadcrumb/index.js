import * as BreadcrumbStyle from "./Breadcrumb.css";
export { BreadcrumbStyle };
