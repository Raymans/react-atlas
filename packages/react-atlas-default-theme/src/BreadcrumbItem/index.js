import * as BreadcrumbItemStyle from "./BreadcrumbItem.css";
export { BreadcrumbItemStyle };
