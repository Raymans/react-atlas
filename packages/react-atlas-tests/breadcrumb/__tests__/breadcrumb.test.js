import React from "react";
import { BreadcrumbCore } from "../../../react-atlas-core/src/Breadcrumb/index";
import { BreadcrumbItemCore } from "../../../react-atlas-core/src/BreadcrumbItem/index";
import { TextCore } from "../../../react-atlas-core/src/Text/index";

import renderer from "react-test-renderer";

describe("Test correct render", () => {
  it("Test correct render", function() {
    const tree = renderer
      .create(
        <BreadcrumbCore>
          <BreadcrumbItemCore>
            <TextCore as="a" href="#">
              Atlas
            </TextCore>
          </BreadcrumbItemCore>
          <BreadcrumbItemCore>
            <TextCore as="a" href="#breadcrumb">
              Breadcrumb
            </TextCore>
          </BreadcrumbItemCore>
          <BreadcrumbItemCore active>
            <TextCore>Breadcrumb Example</TextCore>
          </BreadcrumbItemCore>
        </BreadcrumbCore>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
