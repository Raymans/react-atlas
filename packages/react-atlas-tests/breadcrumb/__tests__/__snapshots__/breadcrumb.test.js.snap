// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test correct render Test correct render 1`] = `
<ol
  aria-label="breadcrumbs"
  className=""
  role="navigation"
  style={undefined}
  styleName="breadcrumb"
>
  <li
    className=""
    style={undefined}
    styleName="breadcrumbItem"
  >
    <a
      className={undefined}
      href="#"
      htmlFor={undefined}
      id={undefined}
      style={undefined}
      styleName="defaultFont link"
    >
      Atlas
    </a>
  </li>
  <li
    className=""
    style={undefined}
    styleName="breadcrumbItem"
  >
    <a
      className={undefined}
      href="#breadcrumb"
      htmlFor={undefined}
      id={undefined}
      style={undefined}
      styleName="defaultFont link"
    >
      Breadcrumb
    </a>
  </li>
  <li
    className=""
    style={undefined}
    styleName="breadcrumbItem active"
  >
    <span
      className={undefined}
      htmlFor={undefined}
      id={undefined}
      style={undefined}
      styleName="defaultFont"
    >
      Breadcrumb Example
    </span>
  </li>
</ol>
`;
